ass2.cpp

```diff
@@ -0,0 +1,209 @@
+#include <iostream>
+#include <sstream>
+
+using namespace std;
+
+// TODO: Make two classes for each task with tests
+// TODO: Make comments for each test case
+
+// Task 1
+// 
+// Given a Binary Tree and a key, write a function that prints
+// all the ancestors of the key in the givenbinary tree.
+//
+
+struct node {
+    
+    int data;
+    struct node* left_child = nullptr;
+    struct node* right_child = nullptr;
+    
+    node(int val) {
+        data = val;
+    }
+};
+
+bool have_ancestors(struct node* root, int node_data) {
+    if (root == nullptr) {
+        return false;
+    }
+    // cout << root -> data << endl;
+    
+    if (root -> data == node_data) {
+        return true;
+    }
+    
+    if (have_ancestors(root -> left_child, node_data) || have_ancestors(root -> right_child, node_data)) {
+        cout << root -> data << " ";
+        return true;
+    }
+    return false;
+}
+
+void test_task1() {
+    
+    // case 1
+    node* root = new node(16);
+    root -> left_child = new node(9);
+    root -> right_child = new node(18);
+    
+    std::stringstream ss;
```

Comment:
Very interesting ! I haven't seen before that you can redirect the output of cout :)
I think it will be more clear if the code on lines 50-59 is in a function (that takes the tree and expected output as parameters)